### PR TITLE
Allow switching the Swift version in expressions

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by augusto2112 — APPROVED** [review]

**Review by JDevlieghere — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp [hunk 1]
```diff
@@ -1358,16 +1358,16 @@ SwiftExpressionParser::ParseAndImport(
...
-  if (uint32_t version = m_expr.Language().version) {
-    invocation.getLangOptions().EffectiveLanguageVersion =
+  if (enable_bare_slash_regex_literals)
+    lang_opts.enableFeature(swift::Feature::BareSlashRegexLiterals);
+  if (uint32_t version = m_expr.Language().version)
+    lang_opts.EffectiveLanguageVersion =
         llvm::VersionTuple(version / 100, version % 100);
-  }
+  if (lang_opts.EffectiveLanguageVersion >= swift::version::Version({6}))
+    lang_opts.StrictConcurrencyLevel = swift::StrictConcurrency::Complete;
```
Comment (kastiglione) [inline lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1370]: does the compiler not do this when the version is set to `>=6`? If lldb is doing this, but the compiler isn't, is that going to cause some issues in some cases?

Comment (kastiglione) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1370]: also, I'm confused by the title. The diff appears to affect `StrictConcurrencyLevel`, and the language version already appears supported.

**Review by kastiglione — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test linux

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/73757
@swift-ci test linux

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1370]: Both excellent questions. I asked, and the compiler sets this flag in an ad-hoc fashion that isn't readily available to reuse. This also seems to be the only flag that behaves this way.

The title is wrong. This just adds a test.

**Pull request merged**